packages/maker/src/api/defineSpt.ts

```diff
@@ -25,6 +26,7 @@ export function defineSharedPropertyType(
     array?: boolean;
     description?: string;
     displayName?: string;
+    valueType?: ApiNameValueTypeReference;
```

Comment:
should we add an invariant that the value type doesn't exist, similar to the SPT already exists?

also - if we provide valueType, we can omit type/array right? Rather than force repeating them

Reply:
Well we collapse them based on version. I guess we should check if you are adding 2 spts at the same version